mllib/src/main/scala/org/apache/spark/ml/feature/QuantileDiscretizer.scala

```diff
@@ -93,7 +93,7 @@ private[feature] trait QuantileDiscretizerBase extends Params
    * @group param
    */
   @Since("2.1.0")
-  override val handleInvalid: Param[String] = new Param[String](this, "handleInvalid",
+  final override val handleInvalid: Param[String] = new Param[String](this, "handleInvalid",
```

Comment:
For this and the followings who added before 2.2, this involves breaking change in a way, but I think we should keep them ```final``` to prevent being changed(as we did for other param variables). cc @MLnick @WeichenXu123

Reply:
I'm more prefer to mark them as non final, since if user want to extend these estimators they would be able to override `handleInvalid` to define their own `handleInvalid` valid values (through `doc` parameter of `Param` class constructor) , this is different from other classes. What do you think of it?

Reply:
Fair enough. I will leave ```handleInvalid``` in all estimators ```non-final```.